python/pyspark/mllib/feature.py

```diff
@@ -447,6 +454,17 @@ class Word2Vec(object):
     >>> syms = model.findSynonyms(vec, 2)
     >>> [s[0] for s in syms]
     [u'b', u'c']
+
+    >>> import os, tempfile
+    >>> path = tempfile.mkdtemp()
+    >>> model.save(sc, path)
+    >>> sameModel = Word2VecModel.load(sc, path)
+    >>> model.transform("a") == sameModel.transform("a")
```

Comment:
It has been advised by @mengxr to keep doctests and testing for correctness separately. (https://github.com/apache/spark/pull/6499#discussion_r32784734)
It might be better to just write sameModel.transform("a") and write a small test to test model and sameModel give same results using `getVectors` maybe?